### PR TITLE
Fix point selection highlight

### PR DESCRIPTION
**Pull request by brisvag**
# Description
Fix some tests for point selection, and fix highlight not being updated correctly when changing zoom.

To reproduce the issue on main:

```py
import napari
import numpy as np
v = napari.Viewer()
pts = np.array([[0, 0], [10, 10]])
pl = v.add_points(pts, scale=[5, 5])
pl.mode = 'select'
```

Then select points (manually or with a box) and zoom in and out, and repeat. You'll see that unless you hover a point, the hihglight thickness is not properly updated with the zoom. This PR ensures that zooming always triggers an update to the highlight thickness.

### CODE REVIEW

**Review by alisterburt — COMMENTED** [review]
Good stuff - could you add a test for get value with a mouse position on and off the point? With a test this is good to go!

**Comment by brisvag** [comment]
fixed and added some test that fails on main. I would still wait with merging this after we get #5582, which will affect some of this stuff.

**Comment by brisvag** [comment]
This was actually already fixed in #5582. This PR should still be merged just for the test improvement and for the fix to highlight updating (which sometimes is not triggering when needed on main).

**Comment by brisvag** [comment]
PS: hopefully a lot of this selection logic will soon be replaced by https://github.com/vispy/vispy/pull/2500!

**Comment by psobolewskiPhD** [comment]
Hmm, I still get wierd zoom issues with the points highlights—maybe related to the macOS point GL size issue? When zoomed out, highlight is visible, but not around the edge:
image
When zoomed in, highlight is lost:
image

**Comment by brisvag** [comment]
Can you give me the exact code and zoom to reach these states? And what happens on main with the same states?

But yeah, it sounds like the usual mac issue :/

**Comment by psobolewskiPhD** [comment]
@brisvag I used your snippet from above.
The visible ring in the middle of the point is with:
```
In [5]: v.camera.zoom
Out[5]: 0.3924111300416685
```
and with increasing zoom gets worse.
By
```
In [6]: v.camera.zoom
Out[6]: 1.0383707814871845
```
The highlight is just barely visible. And with further zoom out disappears.

On main, the zoom = .4 behavior is similar, but after zooming out to 1.03 the highlight is still visible inside the marker:
image
however, as I zoom out it persists, but then if I pan or such disappears.

**Comment by brisvag** [comment]
I guess you point sizes are also misbehaving at the same rate? As in, the highlight starts looking off about the same time as the points stop getting bigger?

For me, this is how it looks like in this PR:

https://github.com/napari/napari/assets/23482191/883513a7-fd8e-440a-810b-c03bd2fdd9ee

Is this considerably worse than main for you?

File: napari/layers/points/_tests/test_points_mouse_bindings.py [hunk 1]
```diff
@@ -51,7 +51,7 @@ def create_known_points_layer_2d():
         testing when needing to guarantee no point is clicked on.
     """
     data = [[1, 3], [8, 4], [10, 10], [15, 4]]
-    known_non_point = [20, 30]
+    known_non_point = [10, 11]
```
Comment (GenevieveBuckley) [inline napari/layers/points/_tests/test_points_mouse_bindings.py:54]: Why is your known non-point different?

File: napari/layers/points/_tests/test_points_mouse_bindings.py [hunk 2]
```diff
@@ -470,6 +470,20 @@ def test_unselecting_points(create_known_points_layer_2d):
     # Check clicked point selected
     assert len(layer.selected_data) == 0
 
+    # check that this also works with scaled data and position near a point (see #5737)
```
Comment (GenevieveBuckley) [inline napari/layers/points/_tests/test_points_mouse_bindings.py:473]: Good idea, I'd like to see this check implemented before merging.

File: napari/layers/points/_tests/test_points_mouse_bindings.py [hunk 3]
```diff
@@ -478,21 +492,25 @@ def test_selecting_all_points_with_drag_2d(create_known_points_layer_2d):
     layer.mode = 'select'
 
     # Simulate click
-    event = read_only_event(type='mouse_press', position=known_non_point)
+    event = read_only_event(type='mouse_press', position=(20, 20))
```
Comment (GenevieveBuckley) [inline napari/layers/points/_tests/test_points_mouse_bindings.py]: Why the change from "`known_non_point`"? This is less readable.

File: napari/layers/points/_tests/test_points_mouse_bindings.py [hunk 4]
```diff
@@ -478,21 +492,25 @@ def test_selecting_all_points_with_drag_2d(create_known_points_layer_2d):
...
     mouse_press_callbacks(layer, event)
 
     # Simulate drag start
     event = read_only_event(
-        type='mouse_move', is_dragging=True, position=known_non_point
+        type='mouse_move', is_dragging=True, position=(20, 20)
```
Comment (GenevieveBuckley) [inline napari/layers/points/_tests/test_points_mouse_bindings.py]: Ditto - Why the change from "`known_non_point`"? This is less readable.

File: napari/layers/points/_tests/test_points_mouse_bindings.py (hunk 1 above)
Comment (GenevieveBuckley) [reply napari/layers/points/_tests/test_points_mouse_bindings.py:54]: Update: ah, I don't think that's important, never mind.

Comment (brisvag) [reply napari/layers/points/_tests/test_points_mouse_bindings.py:54]: Well, for future reference: The point used to be far "outside" the other points bounding box. I moved it in between (and close to one existing point) to ensure that almost-overlappin gpoints would work as intended.

File: napari/layers/points/_tests/test_points_mouse_bindings.py (hunk 2 above)
Comment (brisvag) [reply napari/layers/points/_tests/test_points_mouse_bindings.py:473]: It is, just below this comment :P

File: napari/layers/points/_tests/test_points_mouse_bindings.py (hunk 3 above)
Comment (brisvag) [reply napari/layers/points/_tests/test_points_mouse_bindings.py]: Using `known_not_point` here was not really correct, cause we're really trying to select by dragging a box that covers all the points. It worked before because `known_not_point` was ouside the points (see comment above to why that was changed).

Basically, this tests does not need to click any `non-point`, it needs to click specifically a point beyond the corner of the bounding box... I could name this in a variable to make it clearer maybe?

File: napari/layers/points/_tests/test_points_mouse_bindings.py (hunk 2 above)
Comment (GenevieveBuckley) [reply napari/layers/points/_tests/test_points_mouse_bindings.py:473]: Oh, duh! Sorry 🤣

File: napari/layers/points/_tests/test_points_mouse_bindings.py (hunk 3 above)
Comment (GenevieveBuckley) [reply napari/layers/points/_tests/test_points_mouse_bindings.py]: > I could name this in a variable to make it clearer maybe?

Yeah, that might be nice to do. It's not a blocker to merging, though

**Review by GenevieveBuckley — APPROVED** [review]

File: napari/layers/points/_tests/test_points_mouse_bindings.py (hunk 3 above)
Comment (brisvag) [reply napari/layers/points/_tests/test_points_mouse_bindings.py]: added variables for clarity!
